src/proto.proto

```diff
@@ -51,4 +45,12 @@ message Node {
   AxisAlignedCube bounding_cube = 1;
   PositionEncoding position_encoding = 2;
   int64 num_points = 3;
+  string id = 4;
```

Comment:
Internally we represent a NodeId as (lvl: u8, index: u32). That would be a more efficient schema and closer to our implementation.